src/main/kotlin/no/nav/familie/ef/sak/vedtak/TotrinnskontrollService.kt

```diff
@@ -89,6 +94,20 @@ class TotrinnskontrollService(private val behandlingshistorikkService: Behandlin
             || behandlingStatus == BehandlingStatus.IVERKSETTER_VEDTAK
             || behandlingStatus == BehandlingStatus.OPPRETTET
 
+    fun validerHarGammelGOgKanLagres(saksbehandling: Saksbehandling) {
+
+        val tilkjentYtelse = tilkjentYtelseRepository.findByBehandlingId(saksbehandling.id)
+        val erGammelG = tilkjentYtelse?.grunnbeløpsdato?.isBefore(nyesteGrunnbeløpGyldigFraOgMed) ?: false
+        val starterEtterNyG =
+                tilkjentYtelse?.andelerTilkjentYtelse?.any { it.stønadTom > nyesteGrunnbeløpGyldigFraOgMed } ?: false
+        if (saksbehandling.type == BehandlingType.FØRSTEGANGSBEHANDLING || saksbehandling.type == BehandlingType.REVURDERING) {
```

Comment:
Er ikke litt av problemet at du har "ny" G, men kun på deler av resultatet? 🤔 
For å finne ut om andelene er "riktige" må vi vel nesten kjøre en beregning på alle andeler som har stønadTom > G-beløpsdato og sjekke at ny kalkyle er lik eksisterende andel? 

```
const harGammeltBeløp = tilkjentYtelse?.andelerTilkjentYtelse?
.filter { it.stønadTom > nyesteGrunnbeløpGyldigFraOgMed } 
.any {beregnYtelse(...).first().beløp != it.beløp}
```